google-cloud-bigquery-data_transfer/synth.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to synthesize generated parts of this library."""
+
+import synthtool as s
+import synthtool.gcp as gcp
+import logging
+import re
+
+logging.basicConfig(level=logging.DEBUG)
+
+gapic = gcp.GAPICGenerator()
+
+v1_library = gapic.ruby_library(
+    'bigquery/datatransfer', 'v1', artman_output_name='google-cloud-ruby/google-cloud-bigquerydatatransfer',
+    config_path='artman_bigquerydatatransfer.yaml'
+)
+
+# Copy everything but Gemfile, .gemspec, and Changelog.md
+s.copy(v1_library / 'lib')
+s.copy(v1_library / 'test')
+s.copy(v1_library / 'Rakefile')
+s.copy(v1_library / 'README.md')
+s.copy(v1_library / 'LICENSE')
+s.copy(v1_library / '.gitignore')
+s.copy(v1_library / '.rubocop.yml')
+s.copy(v1_library / '.yardopts')
+
+# https://github.com/googleapis/gapic-generator/issues/2174
+s.replace(
+    'lib/google/cloud/bigquery/data_transfer.rb',
+    'File\.join\(dir, "\.rb"\)',
+    'dir + ".rb"')
+
+# https://github.com/googleapis/gapic-generator/issues/2179
+s.replace(
+    'README.md',
+    'https://cloud.google.com/bigquerydatatransfer',
+    'https://cloud.google.com/bigquery/transfer/')
+
+# https://github.com/googleapis/gapic-generator/issues/2182
```

Comment:
If the generated Rakefile references constants like `DATA_TRANSFER_KEYFILE_JSON` then the generated credentials file will need to match that. Can you open a gapic bug for that?